File: YET/Programação Funcional/prime.hs
isPrime :: Int -> Bool
isPrime n
  | n <= 1 = False
  | otherwise = not (any (\x -> n `mod` x == 0) [2 .. (n - 1)])

main :: IO ()
main = do
  putStrLn "Enter a number to check if it is prime:"
  input <- getLine
  let n = read input :: Int
  if isPrime n
    then putStrLn (show n ++ " its prime")
    else putStrLn (show n ++ " is not prime")

File: YET/Programação Funcional/prime.c
#include <stdio.h>
#include <stdbool.h>

bool isPrime(int n) {
    if (n <= 1) {
        return false;
    }
    for (int i = 2; i < n; i++) {
        if (n % i == 0) {
            return false;
        }
    }
    return true;
}

int main() {
    int n;
    printf("Enter a number to check if it is prime: ");
    scanf("%d", &n);

    if (isPrime(n)) {
        printf("%d is prime\n", n);
    } else {
        printf("%d is not prime\n", n);
    }

    return 0;
}